src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -982,12 +982,12 @@ private MemberSemanticModel CreateMemberModel(CSharpSyntaxNode node)
                         // TODO (tomat): handle misplaced global statements
                         if (node.Parent.Kind() == SyntaxKind.CompilationUnit)
```

Comment:
It feels that the model should be created for all global statements in the same compilation unit at once, rather than for each statement separately.